### PR TITLE
[PRMT-4860] Ignore destinationGp field and don't persist it on inbound requests - repo ODS code now comes from SSM

### PR DESCRIPTION
**Pull request by AndyFlintNHS**

### CODE REVIEW

**Comment by AndyFlintNHS** [comment]
Stopping the destinationGp field being persisted to the TransferTrackerDB on inbound requests

EHR requests sent to the GP2GP messenger require a repositoryOdsCode field. I’ve refactored the code so that the repo ODS code is retrieved from SSM and set in Gp2gpMessengerService.repositoryOdsCode. Previously, the field used was the one present on the repoIncomingEvent. I didn’t like this approach as I don’t feel the service should be reliant on an external request to define what its own ODS code is.

Ignoring the destinationGp field that’s present on the repo incoming requests. Where appropriate, I’ve left these in the integration tests. There are some instances such as the one below in EhrRequestTest.java where the behaviour that is being tested is the mapping from the JSON->→ Java object and in these cases we need to make sure that we’re ignoring the field and creating the objects as expected.

```
private static String getRepoIncomingMessage() {
    final String repoIncomingEvent = """
        {
            "conversationId": "%s",
            "nhsNumber": "%s",
            "nemsMessageId": "%s",
            "sourceGp": "%s",
            "destinationGp": "%s",
            "nemsEventLastUpdated": "%s"
        }
        """;

    return repoIncomingEvent.formatted(
            INBOUND_CONVERSATION_ID.toString().toUpperCase(),
            NHS_NUMBER,
            NEMS_MESSAGE_ID,
            SOURCE_GP,
            DESTINATION_GP,
            NEMS_EVENT_LAST_UPDATED
    );
}
```

File: src/test/java/uk/nhs/prm/repo/ehrtransferservice/utils/TestDataUtility.java [hunk 1]
```diff
@@ -10,16 +10,15 @@ private TestDataUtility() { }
     public static final String NHS_NUMBER = "9798547485";
     public static final String SOURCE_GP = "B45744";
     public static final String NEMS_MESSAGE_ID = "2d74a113-1076-4c63-91bc-e50d232b6a79";
-    public static final String DESTINATION_GP = "A74854";
     public static final String NEMS_EVENT_LAST_UPDATED = "2023-10-09T15:38:03.291499328Z";
+    public static final String REPOSITORY_ODS_CODE = "REPOSITORY_ODS_CODE";
     public static final String REPOSITORY_ASID = "REPOSITORY";
```
Comment (chrisbloe-nhse) [inline src/test/java/uk/nhs/prm/repo/ehrtransferservice/utils/TestDataUtility.java:15]: Why is this not `REPOSITORY_ASID = "REPOSITORY_ASID"`?

File: src/main/java/uk/nhs/prm/repo/ehrtransferservice/repo_incoming/RepoIncomingEvent.java [hunk 2]
```diff
@@ -9,11 +10,11 @@
 @Builder
 @AllArgsConstructor
 @NoArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class RepoIncomingEvent {
     private String nhsNumber;
     private String sourceGp;
     private String nemsMessageId;
-    private String destinationGp;
     private String nemsEventLastUpdated;
```
Comment (chrisbloe-nhse) [inline src/main/java/uk/nhs/prm/repo/ehrtransferservice/repo_incoming/RepoIncomingEvent.java:18]: Do we use this?

**Review by chrisbloe-nhse — CHANGES_REQUESTED** [review]
A few comments, but otherwise happy :)

File: src/main/java/uk/nhs/prm/repo/ehrtransferservice/repo_incoming/RepoIncomingEvent.java (hunk 2 above)
Comment (AndyFlintNHS) [reply src/main/java/uk/nhs/prm/repo/ehrtransferservice/repo_incoming/RepoIncomingEvent.java:18]: We do not. Removed.

File: src/test/java/uk/nhs/prm/repo/ehrtransferservice/utils/TestDataUtility.java (hunk 1 above)
Comment (AndyFlintNHS) [reply src/test/java/uk/nhs/prm/repo/ehrtransferservice/utils/TestDataUtility.java:15]: Changed.

**Pull request merged**